File: LeetCode/Study Plan/Practice Problems/Easy/Remove Duplicates From Sorted Array/Cpp/remove_duplicates_from_sorted_array.cc
#include <iostream>
#include <vector>

typedef std::vector<int> vi;

int remove_duplicates_from_sorted_array(vi v) {
    int i=0, j=0;
    while(i<v.size() && j<v.size()) {
        int curr = v[j];
        while(j<v.size() && v[j] == curr) {
            j++;
        }
        v[i] = v[j-1];
        i++;
    }
    return i;
}

int main() {
    std::cout<<remove_duplicates_from_sorted_array(vi{1,1,2})<<"\n";
}

File: LeetCode/Easy/Remove Duplicates From Sorted Array/Go/remove_duplicates_from_sorted_array.go
package main

import "fmt" 

func remove_duplicates_from_sorted_array(arr []int) []int {
	i,j := 0,0
	n := len(arr)
	for i<n && j<n  {
		curr := arr[j]
		for j<n && curr == arr[j]{
			j++
		}
		arr[i] = arr[j-1]
		i++
	}
	return i
}

func main () {
	fmt.Println(remove_duplicates_from_sorted_array([]int{1,1,2}))
}